src/main/java/mate/jdbc/lib/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.jdbc.lib.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.dao.ManufacturerDao;
+import mate.jdbc.lib.dao.exceptions.DataProcessingException;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String sqlRequestForGetDataFromDb
+            = "SELECT * FROM manufacturers WHERE is_deleted = false";
+    private static final String sqlRequestForAddManufacturerToDb
+            = "INSERT INTO manufacturers(NAME, COUNTRY) VALUES(?, ?)";
+    private static final String sqlRequestForUpdateDataById
+            = "UPDATE manufacturers SET name = (?), country = (?) WHERE id = (?)";
+    private static final String sqlRequestForDeleteById
+            = "UPDATE manufacturers SET is_deleted = true WHERE id = (?)";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connectToDB = ConnectionUtil.getConnection();
+                    PreparedStatement statement = connectToDB
+                            .prepareStatement(sqlRequestForAddManufacturerToDb,
+                                    Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connectToDB = ConnectionUtil.getConnection();
+                 PreparedStatement getByIdStatement = connectToDB
+                         .prepareStatement(sqlRequestForGetDataFromDb + " AND id=(?)")) {
```

Comment:
it would be better to write full query at the beginning of the method
```suggestion
                         .prepareStatement(query)) {
```